tests/api2/test_iscsi_auth_network.py

```diff
@@ -0,0 +1,137 @@
+def target(target_name, groups):
+    target_config = call('iscsi.target.create', {'name': target_name, 'groups': groups})
+    try:
+        yield target_config
+    finally:
+        call('iscsi.target.delete', target_config['id'])
+
+
+@contextlib.contextmanager
+def extent(extent_name, zvol_name=None):
+    zvol_name = zvol_name or extent_name
+    with dataset(zvol_name, {'type': 'VOLUME', 'volsize': 51200, 'volblocksize': '512', 'sparse': True}) as zvol:
+        extent_config = call('iscsi.extent.create', {'name': extent_name, 'disk': f'zvol/{zvol}'})
+        try:
+            yield extent_config
+        finally:
+            call('iscsi.extent.delete', extent_config['id'])
+
+
+@contextlib.contextmanager
+def target_extent(target_id, extent_id, lun_id):
+    target_extent_config = call(
+        'iscsi.targetextent.create', {'target': target_id, 'extent': extent_id, 'lunid': lun_id}
+    )
+    try:
+        yield target_extent_config
+    finally:
+        call('iscsi.targetextent.delete', target_extent_config['id'])
+
+
+@contextlib.contextmanager
+def configured_target_to_extent():
+    with portal() as portal_config:
+        with initiator() as initiator_config:
+            with target(
+                'test-target', groups=[{
+                    'portal': portal_config['id'],
+                    'initiator': initiator_config['id'],
+                    'auth': None,
+                    'authmethod': 'NONE'
+                    }]
+            ) as target_config:
+                with extent('test_extent') as extent_config:
+                    with target_extent(target_config['id'], extent_config['id'], 1):
+                        yield {
+                            'extent': extent_config,
+                            'target': target_config,
+                            'global': call('iscsi.global.config'),
+                            'portal': portal_config,
+                        }
+
+
+@contextlib.contextmanager
+def configure_iscsi_service():
+    with configured_target_to_extent() as iscsi_config:
+        try:
+            call('service.start', 'iscsitarget')
+            yield iscsi_config
+        finally:
+            call('service.stop', 'iscsitarget')
+
+
+@functools.cache
+def get_nas_ip_subnet():
+    for interface in call('interface.query'):
+        for alias in (interface.get('state', {}).get('aliases', [])):
+            if alias['address'] == ip:
+                return alias['netmask']
+    else:
+        raise Exception(f'Unable to determine NAS {ip!r} IP subnet')
+
+
+def check_iscsi_target_login(target_name, portal_ip):
+    try:
+        run_on_runner(['iscsiadm', '-m', 'node', '--targetname', target_name, '--portal', portal_ip, '--login'])
+    except RunOnRunnerException as e:
+        return False
+    else:
+        run_on_runner(['iscsiadm', '-m', 'node', '--targetname', target_name, '--portal', portal_ip, '--logout'])
+        return True
+
+
+def iscsi_login_test_impl(valid):
```

Comment:
You can use `@pytest.mark.parametrize('valid', [False, True])` here and you won't need those two methods below